src/repositories/RequestRepository.ts

```diff
@@ -0,0 +1,287 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+
+/* eslint-disable class-methods-use-this */
+
+import assert from 'assert';
+import BigNumber from 'bignumber.js';
+import { DataTypes, InitOptions, Model } from 'sequelize';
+
+import Request from '../models/Request';
+import Subject from '../observer/Subject';
+import Utils from '../Utils';
+import { MessageModel } from './MessageRepository';
+
+
+/**
+ * An interface, that represents a row for requests(stake/redeem) table.
+ *
+ * See: http://docs.sequelizejs.com/manual/typescript.html#usage
+ */
+class RequestModel extends Model {
+  public readonly requestHash!: string;
+
+  public readonly requestType!: RequestType;
+
+  public readonly messageHash!: string;
+
+  public readonly amount!: BigNumber;
+
+  public readonly beneficiary!: string;
+
+  public readonly gasPrice!: BigNumber;
+
+  public readonly gasLimit!: BigNumber;
+
+  public readonly nonce!: BigNumber;
+
+  public readonly gateway!: string;
+
+  public readonly sender!: string;
+
+  public readonly senderProxy!: string;
+
+  public readonly blockNumber!: BigNumber;
+
+  public readonly createdAt!: Date;
+
+  public readonly updatedAt!: Date;
+}
+
+/**
+ * Request types which facilitator can process
+ */
+export enum RequestType {
+  Stake = 'stake',
+  Redeem = 'redeem',
+}
+
+/**
+ * Stores instances of Request.
+ *
+ * Class enables creation, update and retrieval of Request objects.
+ * On construction it initializes underlying database model.
+ */
+export default class RequestRepository extends Subject<Request> {
+  /* Public Functions */
+
+  /**
+   * Initializes an underlying model and a database table.
+   * Creates database table if it does not exist.
+   */
+  public constructor(initOptions: InitOptions) {
+    super();
+
+    RequestModel.init(
+      {
+        requestHash: {
+          type: DataTypes.STRING,
+          primaryKey: true,
+        },
+        requestType: {
+          type: DataTypes.ENUM({
+            values: [RequestType.Stake, RequestType.Redeem],
+          }),
+          allowNull: false,
+        },
+        blockNumber: {
+          type: DataTypes.BIGINT,
+          allowNull: false,
+          validate: {
+            min: 0,
+          },
+        },
+        messageHash: {
+          type: DataTypes.STRING,
+          allowNull: true,
+          unique: true,
+        },
+        amount: {
+          type: DataTypes.BIGINT,
+          allowNull: false,
+          validate: {
+            min: 0,
+          },
+        },
+        beneficiary: {
+          type: DataTypes.STRING,
+          allowNull: false,
+        },
+        gasPrice: {
```

Comment:
Model has optional argument for this field but here it's required.

Reply:
As discussed created below ticket to handle this:
https://github.com/mosaicdao/facilitator/issues/188